# HG changeset patch
# User Rik <rik@octave.org>
# Date 1541182472 25200
#      Fri Nov 02 11:14:32 2018 -0700
# Node ID 227e02c26617d25a6afa3362c933594ea6895d1f
# Parent  fbc23950b00aaaa4de1aa21b9d1eb2a8b95e8090
Deprecate C++ is_figure in favor of isfigure for readability.

* NEWS: Announce deprecation and replacement.

* graphics.cc (isfigure): New function which check whether input is a figure
handle.
* graphics.cc (is_figure): Add OCTAVE_DEPRECATED decoration to function.
* graphics.cc (xcreatefcn): Replace call to "is_figure" with "isfigure".

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -163,13 +163,14 @@ Summary of important user-visible change
   ** Certain C++ predicate test functions have been deprecated and
      replaced with the same functionality under the equivalent name used
      in the Octave programming language.  These changes make it easier
      for new programmers familiar with Octave to understand code written
      in C++.
 
       Function             | Replacement
       ---------------------|------------------
+      is_figure            | isfigure
       is_hghandle          | ishghandle
 
 ---------------------------------------------------------
 
 See NEWS.4 for old news.
diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -3010,23 +3010,27 @@ ishghandle (const octave_value& val)
   return retval;
 }
 
 OCTAVE_DEPRECATED (5.0, "use 'ishghandle' instead")
 static octave_value is_hghandle (const octave_value& val)
 { return ishghandle (val); }
 
 static bool
-is_figure (double val)
+isfigure (double val)
 {
   graphics_object go = gh_manager::get_object (val);
 
   return go && go.isa ("figure");
 }
 
+OCTAVE_DEPRECATED (5.0, "use 'isfigure' instead")
+static bool is_figure (double val)
+{ return isfigure (val); }
+
 static void
 xcreatefcn (const graphics_handle& h)
 {
   graphics_object go = gh_manager::get_object (h);
   go.get_properties ().execute_createfcn  ();
 }
 
 static void
@@ -12081,17 +12085,17 @@ Undocumented internal function.
 
   if (args.length () == 0)
     print_usage ();
 
   double val = args(0).xdouble_value ("__go_figure__: figure number must be a double value");
 
   octave_value retval;
 
-  if (is_figure (val))
+  if (isfigure (val))
     {
       graphics_handle h = gh_manager::lookup (val);
 
       xset (h, args.splice (0, 1));
 
       retval = h.value ();
     }
   else
